common/views/components/Checkbox/Checkbox.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+
+import React from 'react';
+import styled from 'styled-components';
+import { classNames } from '../../../utils/classnames';
+import Space from '../styled/Space';
+
+const CheckboxLabel = styled.label.attrs({
+  className: classNames({
+    'flex flex--v-center': true,
+  }),
+})``;
+
+const CheckboxBox = styled.span.attrs({
+  className: classNames({
+    CheckboxBox: true,
+    'flex-inline flex--v-center flex--h-center relative': true,
+  }),
+})`
+  width: 1em;
+  height: 1em;
+  border: 1px solid ${props => props.theme.colors.silver};
+  border-radius: 3px;
+  transition: all 400ms ease;
+
+  &:after {
+    content: '';
+    position: absolute;
+    width: 60%;
+    height: 60%;
+    opacity: 0;
+    transition: opacity 400ms ease;
+    background: url('data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%2032%2032%22%3E%0A%20%20%3Cpath%20fill%3D%22%23ffffff%22%20d%3D%22M6%2012l-6%206%2012%2012%2020-20-6-6-14%2014z%22%3E%3C/path%3E%0A%3C/svg%3E%0A');
+  }
+`;
+
+const CheckboxInput = styled.input.attrs({
+  type: 'checkbox',
+})`
+  position: absolute;
+  opacity: 0;
+  z-index: 1;
+  width: 1em;
+  height: 1em;
+
+  &:checked ~ .CheckboxBox {
+    background: ${props => props.theme.colors.green};
+    border-color: ${props => props.theme.colors.green};
+
+    &:after {
+      opacity: 1;
+    }
+  }
+`;
+
+type CheckboxProps = {|
+  id: string,
+|};
+type CheckboxWithLabelProps = {|
+  ...CheckboxProps,
+  text: string,
+|};
+
+export function Checkbox({ id, ...inputProps }: CheckboxProps) {
```

Comment:
Is it valid aria wise to ever have a checkbox without a label?

Reply:
It wouldn't pass validation without one, no, but it is up to the consumer of this named export to supply a `<label htmlFor="id">`.

Reply:
Do we have an instance where we might need to use the invalid implementation?

Reply:
Ha – I thought you might ask that. We have different font styles for the labels at `/checkbox` than those in the filter designs we've been working on. But I guess the newsletter stuff is definitely likely to change, so perhaps we keep the default export for now, and don't mark the `Checkbox` function as a named export until the point it is required as such?